gabi_test.go

```diff
@@ -1150,6 +1158,384 @@ func TestConstructCredentialNonZeroRandomBlindAttributes(t *testing.T) {
 	assert.EqualError(t, err, "attribute at random blind index should be nil before issuance")
 }
 
+func TestKeyshareResponse(t *testing.T) {
+	// Setup secrets and randomizer
+	ourSecret, err := GenerateSecretAttribute()
+	require.NoError(t, err)
+	userSecret, err := GenerateSecretAttribute()
+	require.NoError(t, err)
+
+	// Ensure that the sum of these two is not larger than an ordinary secret key
+	ourSecret.Div(ourSecret, big.NewInt(2))
+	userSecret.Div(userSecret, big.NewInt(2))
+
+	statement, err := rangeproof.NewStatement(rangeproof.GreaterOrEqual, big.NewInt(1))
+	require.NoError(t, err)
+	rangeStatements := map[int][]*rangeproof.Statement{
+		2: {statement},
+	}
+
+	tests := map[string]struct {
+		builders  ProofBuilderList
+		signature bool
+	}{
+		"Disclosure": {builders: ProofBuilderList{
+			newDisclosureBuilder(t, testPrivK, testPubK, userSecret, ourP(ourSecret, testPubK), false, nil)},
+		},
+		"DisclosureRevocation": {builders: ProofBuilderList{
+			newDisclosureBuilder(t, testPrivK, testPubK, userSecret, ourP(ourSecret, testPubK), true, nil),
+		}},
+		"DisclosureRangeProof": {builders: ProofBuilderList{
+			newDisclosureBuilder(t, testPrivK, testPubK, userSecret, ourP(ourSecret, testPubK), false, rangeStatements),
+		}},
+		"DisclosureRevocationRangeProof": {builders: ProofBuilderList{
+			newDisclosureBuilder(t, testPrivK, testPubK, userSecret, ourP(ourSecret, testPubK), true, rangeStatements),
+		}},
+		"DoubleDisclosureSameKeys": {builders: ProofBuilderList{
```

Comment:
Just out of curiosity (and lack of understanding): Do you have an example when such a double disclosure with the same keys would be used?

Reply:
For example, disclosing `pbdf.gemeente.personalData.fullName` and `pbdf.gemeente.address.municipality` in a single disclosure session. That would result in two disclosure builders being used that use the same public key, as is tested here. Not sure if there is currently an actual use case somewhere doing that, but regardless we certainly need to support it.